Tests/Extractor/CoffeeExtractorTest.php

```diff
@@ -0,0 +1,190 @@
+use Symfony\Component\Translation\MessageCatalogue;
+use Bazinga\Bundle\JsTranslationBundle\Finder\FinderFactory;
+use Bazinga\Bundle\JsTranslationBundle\Filesystem\Filesystem;
+use Bazinga\Bundle\JsTranslationBundle\Extractor\CoffeeExtractor;
+
+final class CoffeeExtractorTest extends PHPUnit_Framework_TestCase
+{
+    const TEST_LOCALE = 'en';
+    const TEST_KEY_1 = 'test-key-1';
+    const TRANSLATION_PATH_VIEWS = '/translation-path/views';
+    const TRANSLATION_PATH_PUBLIC = '/translation-path/public';
+
+    /**
+     * @var CoffeeExtractor
+     */
+    private $sut;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var FinderFactory
+     */
+    private $finderFactory;
+
+    /**
+     * @var MessageCatalogue
+     */
+    private $messageCatalogue;
+
+    private $folder;
+    private $fileName;
+    private $fileContent;
+
+    public function setUp()
+    {
+        $this->filesystem = $this->prophesize('Bazinga\Bundle\JsTranslationBundle\Filesystem\Filesystem');
+        $this->finderFactory = $this->prophesize('Bazinga\Bundle\JsTranslationBundle\Finder\FinderFactory');
+        $this->messageCatalogue = new MessageCatalogue(self::TEST_LOCALE);
+
+        $this->sut = new CoffeeExtractor(
+            $this->filesystem->reveal(),
+            $this->finderFactory->reveal()
+        );
+    }
+
+    public function testExtractShouldNotRetrieveTransKey()
+    {
+        $this->givenASourceFolderWithNotValidTransFunctionUsage();
+        $this->thenTheFinderWillFindACoffeeFile();
+        $this->andTheFilesystemWillGrabItsContent();
+        $this->whenUsingTheExtractFromTheSut();
+        $this->assertTheMessageCatalogueIsEmpty();
+    }
+
+    public function testExtractShouldRetrieveTransKey()
+    {
+        $this->givenASourceFolderWithATransFunctionUsage();
+        $this->thenTheFinderWillFindACoffeeFile();
+        $this->andTheFilesystemWillGrabItsContent();
+        $this->whenUsingTheExtractFromTheSut();
+        $this->assertTheTransKeyIsInMessageCatalogue();
+    }
+
+    public function testExtractShouldNotRetrieveTransChoiceKey()
+    {
+        $this->givenASourceFolderWithNotValidTransChoiceFunctionUsage();
+        $this->thenTheFinderWillFindACoffeeFile();
+        $this->andTheFilesystemWillGrabItsContent();
+        $this->whenUsingTheExtractFromTheSut();
+        $this->assertTheMessageCatalogueIsEmpty();
+    }
+
+    public function testExtractShouldRetrieveTransChoiceKey()
+    {
+        $this->givenASourceFolderWithATransChoiceFunctionUsage();
+        $this->thenTheFinderWillFindACoffeeFile();
+        $this->andTheFilesystemWillGrabItsContent();
+        $this->whenUsingTheExtractFromTheSut();
+        $this->assertTheTransChoiceKeyIsInMessageCatalogue();
+    }
+
+    private function givenASourceFolderWithNotValidTransFunctionUsage()
+    {
+        $this->givenASourceFolder();
+
+        $this->fileContent = <<<STRING
+        Translator.tras 'test-key-1'
```

Comment:
@jcchavezs Is there a typo? Should this be `trans` instaed of `tras` ?

Reply:
There is. Sorry.

Reply:
my bad in fact it is not a typo as this test `NotValidTransFunctionUsage`